core/trino-parser/src/main/java/io/trino/sql/parser/ErrorHandler.java

```diff
@@ -301,7 +301,7 @@ else if (ignoredRules.contains(rule)) {
                     if (transition instanceof RuleTransition) {
                         RuleTransition ruleTransition = (RuleTransition) transition;
                         for (int endToken : process(new ParsingState(ruleTransition.target, tokenIndex, suppressed, parser), ruleTransition.precedence)) {
-                            activeStates.push(new ParsingState(ruleTransition.followState, endToken, suppressed, parser));
+                            activeStates.push(new ParsingState(ruleTransition.followState, endToken, false, parser));
```

Comment:
This should continue to suppress if the nested rule did not consume any tokens (i.e., the invoked rule acts a as an epsilon transition)